lib/ddtrace/contrib/sinatra/tracer.rb

```diff
@@ -0,0 +1,165 @@
+
+require 'sinatra/base'
+
+require 'ddtrace/ext/app_types'
+require 'ddtrace/ext/errors'
+require 'ddtrace/ext/http'
+
+module Datadog
+  module Contrib
+    module Sinatra
+      # TracerCfg is used to manipulate the configuration of the Sinatra
+      # tracing extension.
+      class TracerCfg
+        DEFAULT_CFG = {
+          enabled: true,
+          default_service: 'sinatra',
+          tracer: Datadog.tracer,
+          debug: false,
+          trace_agent_hostname: 'localhost',
+          trace_agent_port: 7777
+        }.freeze()
+
+        attr_accessor :cfg, :request_span
+
+        def initialize
+          @cfg = DEFAULT_CFG.dup()
+          @request_span = nil
+        end
+
+        def configure(**args)
+          args.each do |name, value|
+            self[name] = value
+          end
+
+          apply()
+        end
+
+        def apply
+          Datadog::Tracer.debug_logging = @cfg[:debug]
+
+          @cfg[:tracer].enabled = @cfg[:enabled]
+          @cfg[:tracer].configure(hostname: @cfg[:trace_agent_hostname],
+                                  port: @cfg[:trace_agent_port])
+        end
+
+        def [](key)
```

Comment:
We use a `TracerCfg` instance to store the configuration of the tracer, and we install it as a sinatra setting.